Podcast/TabbedPageViewController.swift

```diff
@@ -167,25 +146,95 @@ class TabbedPageViewController: UIViewController, UIPageViewControllerDataSource
     
     
     func updateSearchResults(for searchController: UISearchController) {
-        guard let searchText = searchController.searchBar.text else { return }
-        updateCurrentViewControllerTableView()
+        guard let searchText = searchController.searchBar.text, searchText != "" else { return }
+        print("updateSearchResults", searchText)
+        self.searchText = searchText
+        self.sectionOffsets = [.episodes: 0, .series: 0, .people: 0, .tags: 0]
+        fetchData(type: tabSections[tabBar.selectedIndex], query: searchText, offset: 0, max: PageSize, append: false)
         searchController.searchResultsController?.view.isHidden = false
     }
     
-    func updateCurrentViewControllerTableView() {
-        guard let currentViewController = viewControllers[tabBar.selectedIndex] as? SearchTableViewController else { return }
+    func updateCurrentViewControllerTableView(append: Bool, indexBounds: (Int, Int)?) {
+        if append {
+            guard let currentViewController = viewControllers[tabBar.selectedIndex] as? SearchTableViewController else { return }
+            currentViewController.searchResults = searchResults
+            guard let (start, end) = indexBounds else {
+                currentViewController.tableView.reloadData()
+                return
+            }
+            let indexPaths = (start..<end).map { return IndexPath(row: $0, section: 0) }
+            currentViewController.tableView.beginUpdates()
+            currentViewController.tableView.insertRows(at: indexPaths, with: .automatic)
+            currentViewController.tableView.endUpdates()
+            currentViewController.tableView.finishInfiniteScroll()
+            return
+        }
         
-        switch currentViewController.searchType {
-        case .episodes:
-            break
-        case .series:
-            break
-        case .people:
-            break
-        case .tags:
-            break
+        // If we are not appending then we are fetching data for all tables
+        guard let searchTableViewControllers = viewControllers as? [SearchTableViewController] else { return }
+        for viewController in searchTableViewControllers {
+            viewController.searchResults = searchResults
+            viewController.tableView.reloadData()
         }
-        currentViewController.tableView.reloadData()
+    }
+    
+    func fetchData(type: SearchType, query: String, offset: Int, max: Int, append: Bool) {
+        
+        System.endpointRequestQueue.cancelAllEndpointRequestsOfType(type: SearchEpisodesEndpointRequest.self)
+        System.endpointRequestQueue.cancelAllEndpointRequestsOfType(type: SearchUsersEndpointRequest.self)
+        System.endpointRequestQueue.cancelAllEndpointRequestsOfType(type: SearchSeriesEndpointRequest.self)
+        System.endpointRequestQueue.cancelAllEndpointRequestsOfType(type: SearchAllEndpointRequest.self)
+        
+        if append {
+            switch type {
```

Comment:
why don't you have a .all type so you don't need to use append and can flatten this logic?